### ISSUE TITLE
How to use undefined in property expression?

### ISSUE THREAD

**Issue by tomager** [pureqml/qmlcore#228]
I'd like to say something like

```
Text {
  text: rp.value === undefined ? "" : rp.value;
}
```

but that fails with `src.q1: error: Property src.<anonymous>.undefined could not be resolved`. Similarly,

```
Text {
  text: this._strip_undefined(rp.value);
}
```

fails because 'this' cannot be resolved.

What is the normal/expected way to do this? Sorry if I've missed something obvious/basic.

**Comment by whoozle** [comment]
are you using `property var` for that? interesting, we don't have any `undefined` in expressions in any project :see_no_evil: 

undefined fixed in 3b2cbdf4b9b57ab2e9b5dc7acadaffcc089e0ebc

"this" fix will arrive shortly.

**Comment by tomager** [comment]
That's awesome-- thank you!

**Comment by whoozle** [comment]
fixed this in ae229afc64a424f3fa4390f548808cf53fd25f1f, but be careful - this.XXX won't add depedency to XXX

**Comment by whoozle** [comment]
fixed this in ae229afc64a424f3fa4390f548808cf53fd25f1f, but be careful - this.XXX won't add depedency to XXX
